dashboard/app/models/lesson.rb

```diff
@@ -723,6 +723,88 @@ def lesson_plan_has_verified_resources
     resources.any? {|r| r.audience == 'Verified Teacher'}
   end
 
+  def self.copy_to_script(original_lesson, destination_script)
+    return if original_lesson.script == destination_script
+    raise 'Both lesson and script must be migrated' unless original_lesson.script.is_migrated? && destination_script.is_migrated?
+    raise 'Destination script must be in a course version' if destination_script.get_course_version.nil?
+
+    ActiveRecord::Base.transaction do
+      copied_lesson = original_lesson.dup
+      copied_lesson.key = copied_lesson.name
```

Comment:
Do we have any rules on the style of lesson keys? Should we? Should we try to turn the name into something that is all lower case and has dashes or underscores here for key?

Reply:
Unless there is a specific benefit I'm not seeing, I think this is probably more trouble than it's worth.

Reply:
fwiw the reason I set the key here is because some lessons have a key like `lesson_2` which would then conflict with a lesson already in the script. I figured the name was normally unique so I used it but it still feels a bit weird